### DISCUSSION TITLE
Whether DI-star training code open source？

### DISCUSSION THREAD

**Discussion by zhangjingwei713** [opendilab/DI-engine#65] [Q&A] [answered]
DI-star imitation learning and reinforcement learning training process code for example how to give rewards, where the unit should move to, etc. Whether this part of the code open source or not? Is each unit trained to be a individual agent?

**Comment by liuyuisanai** [comment, marked as answer]
> DI-star imitation learning and reinforcement learning training process code for example how to give rewards, where the unit should move to, etc. 

>Whether this part of the code open source or not? 

Hi @zhangjingwei713 , Yes the training code will be released. However, currently the DI-star team is working on another project so we have to postpone the release date of the whole project to 2021 Q4. Due to the huge difference between our internal training framework vs. DI-engine, it takes some time for us to prepare.  Pls stay tuned and more progress will be coming soon. @upia99 please track this issue.
>Is each unit trained to be a individual agent?

No, all units and their actions are controlled by a single agent. Pls refer to the publication of AlphaStar.
